### ISSUE TITLE
Circuit-Breakers not timing out even though documentation states they should

### ISSUE THREAD

**Issue by guess-burger** [sunng87/diehard#69]
The documentation for circuit-breakers it states that it's possible to add a timeout as part of the circuit-breaker definition. Here's the relevant part of the documentation.
```
(repl/doc dh/defcircuitbreaker)
-------------------------
diehard.core/defcircuitbreaker
([name opts])
Macro
  Define a circuit breaker with option.
#### Available options

There options are available when creating circuit breaker in
`defcircuitbreaker`.

##### Failure criteria

All the three `fail` options share same meaning with similar option in
retry block.

* `:fail-if`
* `:fail-on`
* `:fail-when`
* `:timeout-ms` while give all you code a timeout is best practice in
  application level, circuit breaker also provides a timeout for
  marking a long running block as failure
...
...
```
However, when using `:timeout-ms` with a circuit-breaker the timeout does not fire.
```
(dh/defcircuitbreaker my-cb {:failure-threshold-ratio [8 10]
                             :delay-ms 1000
                             :timeout-ms 1000})
;=> #'user/my-cb

(dh/with-circuit-breaker my-cb
  (do (Thread/sleep 2000)
      :should-not-see-this))
;=> :should-not-see-this

(dh/with-timeout {:timeout-ms 1000}
  (do (Thread/sleep 2000)
      :should-not-see-this))
Execution error (TimeoutExceededException) at dev.failsafe.internal.TimeoutExecutor/lambda$apply$0 (TimeoutExecutor.java:73).
null
```
Additionally, the only reference to `:timeout-ms` in `circuit-breaker.clj` I can see is it's mention in [`allowed-cricuit-breaker-option-keys`](https://github.com/sunng87/diehard/blob/master/src/diehard/circuit_breaker.clj#L13C15-L13C26).

Should timeouts be working or is that documentation an error?

**Comment by sunng87** [comment]
@guess-burger Thank you for reporting. This should be a documentation error. IIRC, it was an option supported in early version and now being removed.

**Comment by sunng87** [comment]
fixed in [1af9762](https://github.com/sunng87/diehard/commit/1af9762264acf2efbae29bf9578efd420ec32549)
